src/rgw/rgw_data_sync.cc

```diff
@@ -1230,10 +1244,13 @@ class RGWDataSyncCR : public RGWCoroutine {
                  iter != sync_status.sync_markers.end(); ++iter) {
               RGWDataSyncShardControlCR *cr = new RGWDataSyncShardControlCR(sync_env, sync_env->store->get_zone_params().log_pool,
                                                         iter->first, iter->second);
+              cr->get();
               shard_crs_lock.Lock();
               shard_crs[iter->first] = cr;
               shard_crs_lock.Unlock();
-              spawn(cr, true);
+              RGWCoroutinesStack *stack = spawn(cr, true);
+              stack->get();
+              stacks[stack] = iter->first;
```

Comment:
I'm not sure what `map<RGWCoroutinesStack *, int> stacks` accomplishes here (I don't see it being accessed elsewhere), and this reference from `stack->get()` isn't ever released.